### DISCUSSION TITLE
Windows 开发环境无法通过 eslint：import/no-unresolved

### DISCUSSION THREAD

**Discussion by inkss** [ArtalkJS/Artalk#621] [Q&A] [answered]
![23-10-28_235231](https://github.com/ArtalkJS/Artalk/assets/31947043/4c7dd746-8952-44e6-82fe-f0312b5efa10)

**Comment by qwqcode** [comment]
你用的是pnpm吗

**Reply by qwqcode** [reply]
奇怪 🤔 试试删掉 node_modules 目录然后再 pnpm install 呢。

我这里能通过 eslint 检测（是 macOS 环境），后有个 CI 是 Debian Linux 环境也能通过编译

可能是 Windows 上面某个依赖有问题？

这错误和 tsconfig.json 的 paths 有关（ui/packages/artalk/tsconfig.json）应该还和 ui/.eslintrc.js 的 import plugins 有关

**Reply by qwqcode** [reply]
最近对前端代码进行了大量重构（功能进一步更细地拆分），但还没完工，有很多功能有待测试和修bug 🤣

**Reply by inkss** [reply]
我也超级迷惑了

**Reply by qwqcode** [reply, marked as answer]
把 `eslint-import-resolver-typescript` 依赖降级到 `2.7.0` 就正常了 🤣

我看到它的 [Release notes](https://github.com/import-js/eslint-import-resolver-typescript/releases/tag/v3.0.0)：use get-tsconfig to replace tsconfig-paths ([78a08e0](https://github.com/alexgorbatchev/eslint-import-resolver-typescript/commit/78a08e082dcd5ff9e3490759f4919316e715d3ff))

可能新的 https://github.com/privatenumber/get-tsconfig 在 eslint-import-resolver-typescript 中的集成在 win 环境下有 bug。暂时还没明确具体原因。但是降级到 `2.7.0` 就能跑了

**ui\package.json**

```diff
-    "eslint-import-resolver-typescript": "3.6.0",
+    "eslint-import-resolver-typescript": "2.7.0",
```

**Reply by inkss** [reply]
确实！
![02D5A35D](https://github.com/ArtalkJS/Artalk/assets/31947043/f5f56d25-8d31-4c1e-8c12-4a5b58ec5761)

**Comment by inkss** [comment]
直接问你是最快的，emmmm：

```
TypeError: Cannot read properties of undefined (reading 'version')，无法获取评论列表数据
```

![23-10-30_104008](https://github.com/ArtalkJS/Artalk/assets/31947043/f3d9b9c1-1363-4ca7-bad6-9110ddae392e)
![23-10-30_104129](https://github.com/ArtalkJS/Artalk/assets/31947043/e843907a-56eb-4410-a5a2-8790d4f0f5db)

我就随便先填个数字得了..

**Reply by inkss** [reply]
后端版本我是下的编译版的，不知道为什么没带版本号。

![23-10-30_104712](https://github.com/ArtalkJS/Artalk/assets/31947043/dc0d1105-7299-43c7-9962-af48d82f9993)

**Reply by qwqcode** [reply]
需要执行 git pull --tags，然后 make build、或 make dev 才能带上版本号

相关代码：https://github.com/ArtalkJS/Artalk/blob/master/Makefile

**Reply by inkss** [reply]
后端是从github上下的，没编译，go语言完全不懂

**Reply by qwqcode** [reply]
没事 命令行直接执行 make dev 后端就能跑起来了
